src/software/ai/evaluation/time_to_travel_test.cpp

```diff
@@ -0,0 +1,184 @@
+
+    double deceleration_time = (final_vel - max_vel) / -max_accel;
+    // d = (Vi + Vf) / 2 * t
+    double deceleration_distance = (max_vel + final_vel) / 2 * deceleration_time;
+
+    double time_at_max_vel =
+        (distance - acceleration_distance - deceleration_distance) / max_vel;
+
+    double travel_time_expected = acceleration_time + time_at_max_vel + deceleration_time;
+
+    EXPECT_TRUE(TestUtil::equalWithinTolerance(
+        Duration::fromSeconds(travel_time_expected),
+        getTimeToTravelDistance(distance, max_vel, max_accel, initial_vel, final_vel)));
+}
+
+TEST_F(
+    TimeToTravel,
+    getTimeToTravelDistance_with_final_velocity_not_achievable_in_distance_and_greater_than_initial_velocity)
+{
+    // Check that the robot reaches the dest with in an expected time frame when
+    // it does not have enough space to accelerate to the final velocity
+
+    // Equal initial and final velocities, both in the direction of the destination
+    double initial_vel = 0.0;
+    double final_vel   = 4.0;
+    double max_accel   = 1.0;
+    double max_vel     = 4.0;
+    double distance    = 1.0;
+
+    // The robot is accelerating throughout the path (since the final velocity is too
+    // great for it to reach within the distance) t = (-Vi + sqrt(Vi^2 + 2 * a * d)) / a
+    double acceleration_time =
+        (-initial_vel + std::sqrt(std::pow(initial_vel, 2) + 2 * max_accel * distance)) /
+        max_accel;
+
+    // For the upper bound, just choose a time that's much greater then we would expect
+    double max_time_to_dest = 5.0;
+
+    Duration t =
+        getTimeToTravelDistance(distance, max_vel, max_accel, initial_vel, final_vel);
+    EXPECT_LE(Duration::fromSeconds(acceleration_time), t);
+    EXPECT_GE(Duration::fromSeconds(max_time_to_dest), t);
+}
+
+TEST_F(
+    TimeToTravel,
+    getTimeToPositionForRobot_with_final_velocity_not_achievable_in_distance_and_smaller_than_initial_velocity)
+{
+    // Check that the robot reaches the dest with in an expected time frame when
+    // it does not have enough space to accelerate to the final velocity
+
+    // Equal initial and final velocities, both in the direction of the destination
+    double initial_vel = 4.0;
+    double final_vel   = 0.0;
+    double max_accel   = 1.0;
+    double max_vel     = 4.0;
+    double dist_required_to_reach_v_f =
+        std::abs(std::pow(final_vel, 2) - std::pow(initial_vel, 2)) / (2 * max_accel);
+    double distance = dist_required_to_reach_v_f - 0.00001;
+    std::cout << "dist=" << distance << std::endl;
+
+    // The robot is decelerating throughout the path (since the final velocity is too
+    // small for it to reach within the distance) t = (-Vi + sqrt(Vi^2 + 2 * a * d)) / a,
+    // with a being negative due to deceleration
+    //    double acceleration_time =
+    //        (-initial_vel + std::sqrt(std::pow(initial_vel, 2) - 2 * max_accel *
+    //        distance)) / -max_accel;
```

Comment:
you're good to remove these lines from the code since you added the formula in comments above this anyway

Reply:
Good catch. I've removed it